### PR TITLE
[TS] Migrate DataChart tests to tsx

### PR DESCRIPTION
**Pull request by alexmirrington**
#### What does this PR do?

This PR migrates `DataChart` component tests from `*.js` to `*.tsx` to allow `ts-jest` to catch type-related errors in the unit testing pipeline. A few changes to the tests had to be made to support this, listed below:

- Extract values from `.map(...)` operations to constants to allow the addition of explicit typings.
- Fix some type mismatches in the tests themselves
- Relax some fields in `src/js/components/DataChart/index.d.ts` to match things we are already testing for in test cases.

#### Where should the reviewer start?

- Ensure all tests are currently passing
- Refer to the background context section below for more information about the initial errors picked up by TypeScript and the fixed implemented in this PR.

#### What testing has been done on this PR?

- Manually ran and passed unit tests locally for the changed component
- CircleCI tests have passed.

#### How should this be manually tested?

- The reviewer should verify all changes made to tests and ensure that:
    - All test removals are due to unused legacy code and will not cause regressions
    - All tests behave identically and modifications are type-related.

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### Any background context you want to provide?

None

#### What are the relevant issues?

Parent tracking issue: #5599

#### Do the grommet docs need to be updated?

No

#### Should this PR be mentioned in the release notes?

No, all fixes are test-related and not component-related.

#### Is this change backwards compatible or is it a breaking change?

Backwards compatible.

### CODE REVIEW

File: src/js/components/DataChart/__tests__/DataChart-test.tsx [hunk 1]
```diff
@@ -5,15 +5,16 @@ import 'jest-styled-components';
...
+import { ChartProps } from '../../Chart';
 
 const data = [
   { a: 1, b: 'one', c: 111111, d: '2020-06-24' },
   { a: 2, b: 'two', c: 222222, d: '2020-06-23' },
 ];
 
 describe('DataChart', () => {
-  let warnSpy;
+  let warnSpy: jest.SpyInstance;
```
Comment (jcfilben) [inline src/js/components/DataChart/__tests__/DataChart-test.tsx:17]: I'm seeing the following error on this line: `Namespace 'jest' has no exported member 'SpyInstance'.`

If you're aren't seeing this error can you make sure your branch is updated with master and then run `yarn install` again? We recently did some jest dependency upgrades

Comment (alexmirrington) [reply src/js/components/DataChart/__tests__/DataChart-test.tsx:17]: Just ran `yarn install` again, are you sure you have `@types/jest` installed?

Comment (jcfilben) [reply src/js/components/DataChart/__tests__/DataChart-test.tsx:17]: Just checked and I have v29.1.0 of `@types/jest` installed. I can check tomorrow if someone else on my team is seeing the error or not and then look into it more from there

Comment (alexmirrington) [reply src/js/components/DataChart/__tests__/DataChart-test.tsx:17]: Interesting, just verified I have the same version of `@types/jest`, FWIW I'm on node `v18.10.0`

Comment (jcfilben) [reply src/js/components/DataChart/__tests__/DataChart-test.tsx:17]: I tested this with some other people today but we were getting different behavior across our environments. One of us was seeing the error, one was not, and the other was getting a different error. I am on node v14.9.2 so I am going to try and upgrade to amore recent version of node and see if that changes anything.

Comment (jcfilben) [reply src/js/components/DataChart/__tests__/DataChart-test.tsx:17]: I had my visual studio code using a different version of typescript than the project. I changed to use the project version and all looks good now. Sorry for the delay on this!

**Review by jcfilben — APPROVED** [review]
Looks good!

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**